File: src/entity/score.rs
use chrono::NaiveDateTime;
use serde::{Deserialize, Serialize};
use sqlx::{FromRow, types::Uuid};

#[derive(Debug, FromRow, Serialize, Deserialize)]
pub struct Score {
    pub id: Uuid,
    pub username: String,
    pub score: i32,
    pub is_hidden: bool,
    pub game_id: Uuid,
    pub created_at: NaiveDateTime,
    pub updated_at: Option<NaiveDateTime>,
}

File: frontend/src/lib/Models/Score.ts
import type { Level } from "./Level"
import type { User } from "./User"

export interface Score {
    id: string,
    score: number,
    is_hidden: boolean,
    level: Level,
    user: User,
    username: string,
    created_at: string,
    updated_at: string
}

export interface ScoreDTO {
    username: string,
    score: number,
    is_hidden: boolean,
    level_id: string,
    user_id: string
}